website/content/docs/v0.9/Guides/converting-control-plane.md

```diff
@@ -0,0 +1,247 @@
+```bash
+$ talosctl -n <CONTROL_PLANE_IP> get bs
+NODE         NAMESPACE   TYPE              ID              VERSION   SELF HOSTED
+172.20.0.2   runtime     BootstrapStatus   control-plane   2         true
+```
+
+Talos machine configuration should be updated to the 0.9 format, there are two new required machine configuration settings:
+
+* `.cluster.serviceAccount` is the service account PEM-encoded private key.
+* `.cluster.aggregatorCA` is the aggregator CA for `kube-apiserver` (certficiate and private key).
+
+Current service account can be fetched from the Kubernetes secrets:
+
+```bash
+$ kubectl -n kube-system get secrets kube-controller-manager -o jsonpath='{.data.service\-account\.key}'
+LS0tLS1CRUdJTiBSU0EgUFJJVkFURS...
+```
+
+All control plane nodes machine configuration should be patched with the service account key:
+
+```bash
+$ talosctl -n <CONTROL_PLANE_IP1>,<CONTROL_PLANE_IP2>,... patch mc --immediate -p '[{"op": "add", "path": "/cluster/serviceAccount", "value": {"key": "LS0tLS1CRUdJTiBSU0EgUFJJVkFURS..."}}]'
+patched mc at the node 172.20.0.2
+```
+
+Aggregator CA can be generated using OpenSSL or any other certificate generation tools: RSA or ECDSA certificate with CN `front-proxy` valid for 10 years.
+PEM-encoded CA certificate and key should be base64-encoded and patched into the machine config at path `/cluster/aggregatorCA`:
+
+```bash
+$ talosctl -n <CONTROL_PLANE_IP1>,<CONTROL_PLANE_IP2>,... patch mc --immediate -p '[{"op": "add", "path": "/cluster/aggregatorCA", "value": {"crt": "S0tLS1CRUdJTiBDRVJUSUZJQ...", "key": "LS0tLS1CRUdJTiBFQy..."}}]'
+patched mc at the node 172.20.0.2
+```
+
+At this point static pod definitions and bootstrap manifests should be rendered, please see "Automated Conversion" on how to verify generated objects.
+Machine configuration can be updated until generated static pod definitions and bootstrap manifests look good.
+
+If static pod definitions are not generated, check logs with `talosctl -n <IP> logs controller-runtime`.
+
+Disable `pod-checkpointer` with:
+
+```bash
+$ kubectl -n kube-system delete ds pod-checkpointer
+daemonset.apps "pod-checkpointer" deleted
+```
+
+Wait for all pod checkpoints to be removed:
+
+```bash
+$ kubectl -n kube-system get pods
+NAME                                            READY   STATUS    RESTARTS   AGE
+...
+pod-checkpointer-8q2lh-talos-default-master-2   1/1     Running   0          3m34s
+pod-checkpointer-nnm5w-talos-default-master-3   1/1     Running   0          3m24s
+pod-checkpointer-qnmdt-talos-default-master-1   1/1     Running   0          2m21s
+```
+
+Pod checkpoints have annotation `checkpointer.alpha.coreos.com/checkpoint-of`.
+
+Once all the pod checkpoints are removed (it takes 5 minutes for the checkpoints to be removed), proceed by removing self-hosted initialized key:
+
+```bash
+talosctl -n <CONTROL_PLANE_IP> convert-k8s --remove-initialized-key
+```
+
+Talos controllers now render static pod definitions, kubelet launches static pods.
```

Comment:
"Talos controllers will now render static pod definitions, and the kubelet will launch any resulting static Pods."